ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageImpl.java

```diff
@@ -41,6 +41,41 @@ public enum StartupOption {
     FORMAT
   }
 
+  public static Builder newBuilder() {
+    return new RaftStorageImpl.Builder();
+  }
+
+  public static class Builder {
+    private File dir;
+    private CorruptionPolicy logCorruptionPolicy;
+    private StartupOption option;
+    private long storageFeeSpaceMin;
+
+    public Builder setDir(File dir) {
+      this.dir = dir;
+      return this;
+    }
+
+    public Builder setLogCorruptionPolicy(CorruptionPolicy logCorruptionPolicy) {
+      this.logCorruptionPolicy = logCorruptionPolicy;
+      return this;
+    }
+
+    public Builder setOption(StartupOption option) {
```

Comment:
We should move StartupOption from RaftStorageImpl to RaftStorage so that it becomes a public API.

Also, let's add a new StartupOption.RECOVER instead of using null.